### DISCUSSION TITLE
How is the binary release of Ipopt compiled on Windows?

### DISCUSSION THREAD

**Discussion by metab0t** [coin-or/Ipopt#807] [General]
I want to replicate the official binary release on my Windows machine. Is there any reproducible build script?

**Comment by svigerske** [comment]
Not so far. It's somewhat following https://coin-or.github.io/Ipopt/INSTALL.html#SYSTEMPACKAGES_MSYS for the setup and doing `configure`, `make`, and `make install` for `ThirdParty-ASL`, `ThirdParty-Mumps`, and `Ipopt`. 
All `configure` calls are with the same `--prefix` and there is a `/path/to/prefix/share/config.site` that looks similar to
```sh
source /etc/config.site

enable_relocatable=yes
enable_msvc=yes

with_metis_lflags="/path/to/metis.lib"
with_metis_cflags="-I/path/to/metis/include/"

MKLDIR=/path/to/mkl-static
with_lapack="$MKLDIR/mkl_core.lib $MKLDIR/mkl_intel_lp64.lib $MKLDIR/mkl_intel_thread.lib libiomp5md.lib"

case $PWD in 
  *ThirdParty-ASL* ) enable_shared=no
  ;;
esac
```

**Reply by svigerske** [reply]
The classic Intel compilers used are no longer offered by Intel, and configure isn't setup so far to work with the new Intel compilers, so it could be difficult to reproduce. If you want to omit Mumps (which needs a Fortran compiler), then doing the same with MS C/C++ compilers could work.
